src/flexmeasures_client/client.py

```diff
@@ -52,27 +54,37 @@ async def request(
         if self.session is None:
             self.session = ClientSession()
 
-        try:
-            async with async_timeout.timeout(self.request_timeout):
-                response = await self.session.request(
-                    url=url,
-                    method=method,
-                    params=params,
-                    headers=headers,
-                    json=json,
-                    ssl=False,
-                )
-                response.raise_for_status()
-        except asyncio.TimeoutError as exception:
-            msg = "Timeout occurred while connecting to the API."
-            raise ConnectionError(
-                msg,
-            ) from exception
-        except (ClientError, socket.gaierror) as exception:
-            msg = "Error occurred while communicating with the API."
-            raise ConnectionError(
-                msg,
-            ) from exception
+        retry_function = lambda x, y: getattr(x, "status") == 400 and (
+            "Scheduling job waiting" in y.get("message", "") or "Scheduling job in progress" in y.get("message", "")
+        )
+
+        while True:
+            try:
+                async with async_timeout.timeout(self.request_timeout):
+                    response = await self.session.request(
+                        method=method,
+                        url=url,
+                        params=params,
+                        headers=headers,
+                        json=json,
+                        ssl=False if "localhost" in self.host else True,
+                    )
+                    payload = await response.json()
+                    response.raise_for_status()
+                    break
+            except asyncio.TimeoutError as exception:
+                msg = "Timeout occurred while connecting to the API."
+                raise ConnectionError(
+                    msg,
+                ) from exception
+            except (ClientError, socket.gaierror) as exception:
+                if retry_function(exception, payload):
```

Comment:
Here I would change the function name to something like `check_retry_condition` because it might be misunderstood as a funtion that actually runs the retry.